test/sms.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var assert = require('assert');
+var request = require('supertest');
+
+var app = require('../lib/app')();
+
+var mocks = require('./lib/mocks');
+
+var UID = 'abcdef123456';
+
+
+describe('/sms', function () {
+  it('forwards properly-authenticated requests through to basket', function (done) {
+    var PHONE_NUMBER = '15555551234';
+    var MSG_ID = 'SMS_Android';
+    var OPTIN = 'N';
+    mocks.mockOAuthResponse().reply(200, {
+      user: UID,
+      scope: 'basket:write'
+    });
+    mocks.mockProfileResponse().reply(200, {
+      email: 'dont@ca.re',
+    });
+    mocks.mockBasketResponse().post('/subscribe_sms/', function (body) {
+      assert.deepEqual(body, {
+        /*eslint-disable camelcase*/
+        mobile_number: PHONE_NUMBER,
+        msg_id: MSG_ID,
+        /*eslint-enable camelcase*/
+        optin: OPTIN
+      });
+      return true;
+    }).reply(200, {
+      status: 'ok',
+    });
+    request(app)
+      .post('/subscribe_sms')
+      .set('authorization', 'Bearer TOKEN')
+      .send({
+        /*eslint-disable camelcase*/
+        mobile_number: PHONE_NUMBER,
+        msg_id: MSG_ID,
+        /*eslint-enable camelcase*/
+        optin: OPTIN
```

Comment:
I believe if you were to optin, you would receive more marketing-based text messages. It's default is `'N'` in the basket repo, and my PR to the content-server explicitly sends `'N'` as well. This proxy I felt shouldn't have an opinion: it's just a dump pipe.